table/src/main/java/org/enso/table/data/column/Storage.java

```diff
@@ -0,0 +1,29 @@
+package org.enso.table.data.column;
+
+/** An abstract representation of a data column. */
+public abstract class Storage {
+  /** @return the number of elements in this column (including NAs) */
+  public abstract long size();
+
+  /** @return the type tag of this column's storage. Must be one of {@link Type} */
+  public abstract long getType();
+
+  /**
+   * Checks whether the value at {@code idx} is missing.
+   *
+   * @param idx the index to check.
+   * @return whether or not the value is missing.
+   */
+  public abstract boolean isNa(long idx);
+
+  /**
+   * Enumerating possible storage types.
+   *
+   * <p>Keep in sync with variables in {@code Table.Table}.
+   */
+  public static final class Type {
+    public static final long LONG = 1;
+    public static final long DOUBLE = 2;
+    public static final long STRING = 3;
```

Comment:
Can we not read these values in `Table.Table` via polyglot?

Reply:
We could, but that depends on layers of inlining to make these constant. So I'd rather have this. Keep in mind this is all very provisional, as I have very little idea of what I'm doing :P As said in the PR description, all of this might as well get rewritten next PR.

Reply:
Well then please note that in a comment next to these.